packages/manager/apps/sap-features-hub/src/pages/installation/stepSystemInformation/InstallationStepSystemInformation.page.tsx

```diff
@@ -0,0 +1,134 @@
+import { useFormSteps } from '@/hooks/formStep/useFormSteps';
+import { useLocalStorage } from '@/hooks/localStorage/useLocalStorage';
+import {
+  SYSTEM_INITIAL_VALUES,
+  SYSTEM_INITIAL_ERRORS,
+  FORM_SAP_SIDS_LABEL,
+  SYSTEM_PASSWORD_INPUTS,
+  SYSTEM_TEXT_INPUTS,
+} from './installationStepSystemInformation.constants';
+
+export default function InstallationStepSystemInformation() {
+  const { t } = useTranslation('installation');
+  const [values, setValues] = useState(SYSTEM_INITIAL_VALUES);
+  const [errors, setErrors] = useState(SYSTEM_INITIAL_ERRORS);
+  const { previousStep, nextStep, currentStepLabel } = useFormSteps();
+  const { setStorageItem } = useLocalStorage();
+
+  const isFormValid = useMemo(
+    () =>
+      Object.values(values).every((value) => !!value) &&
+      Object.values(errors).every((error) => !error),
+    [values, errors],
+  );
+
+  const handleChange = (e: OdsInputChangeEvent) => {
+    const { name, value } = e.detail;
+    setValues((prev) => ({ ...prev, [name]: value }));
+    setErrors((prev) => ({ ...prev, [name]: !e.detail.validity?.valid }));
+  };
+  const handleSubmit = () => {
+    setStorageItem(currentStepLabel, {
+      sids: {
+        sapSid: values.sap_sid,
+        sapHanaSid: values.sap_hana_sid,
+      },
+      passwords: {
+        masterSap: values.sap_master_pwd,
+        masterSapHana: values.sap_hana_master_pwd,
+        sidadm: values.sid_adm_pwd,
+        system: values.sap_hana_system_pwd,
+      },
+    });
+    nextStep();
+  };
+
+  return (
+    <div>
+      <FormTitle title={t('system_title')} subtitle={t('system_subtitle')} />
+      <form className="flex flex-col gap-y-6">
+        <OdsText preset="heading-3">{FORM_SAP_SIDS_LABEL}</OdsText>
+        {SYSTEM_TEXT_INPUTS.map((input) => (
+          <OdsFormField
+            key={input.name}
+            className="w-full max-w-[304px]"
+            error={
+              !!values[input.name] && errors[input.name]
+                ? t('system_sid_validation_message')
+                : undefined
+            }
+          >
+            <label htmlFor={input.name} slot="label">
+              <OdsText>{input.label}</OdsText>
+            </label>
+            <OdsInput
+              name={input.name}
+              id={input.name}
+              placeholder={input.placeholder}
+              className="w-full"
+              pattern={input.pattern}
+              onOdsChange={handleChange}
+              hasError={!!values[input.name] && !!errors[input.name]}
+              value={values[input.name]}
+            />
+            {input.helperKey && (
+              <OdsText slot="helper">{t(input.helperKey)}</OdsText>
+            )}
+          </OdsFormField>
+        ))}
+        <OdsText preset="heading-3">{t('passwords')}</OdsText>
+        {SYSTEM_PASSWORD_INPUTS.map((input) => (
+          <OdsFormField
+            key={input.name}
+            className="w-full max-w-[304px]"
```

Comment:
```suggestion
            className="w-full max-w-[10em]"
```

Or equivalent of 300px with em type ?